plugins/readtimer.koplugin/main.lua

```diff
@@ -66,26 +126,109 @@ function ReadTimer:remaining()
+    self.ui.crelistener:addAdditionalHeaderContent(self.additional_header_content_func)
+    self:update_status_bars(-1)
+end
+function ReadTimer:addAdditionalFooterContent()
+    self.ui.view.footer:addAdditionalFooterContent(self.additional_footer_content_func)
+    self:update_status_bars(-1)
+end
+
+function ReadTimer:removeAdditionalHeaderContent()
+    self.ui.crelistener:removeAdditionalHeaderContent(self.additional_header_content_func)
+    self:update_status_bars(-1)
+    UIManager:broadcastEvent(Event:new("UpdateHeader"))
+end
+function ReadTimer:removeAdditionalFooterContent()
+    self.ui.view.footer:removeAdditionalFooterContent(self.additional_footer_content_func)
+    self:update_status_bars(-1)
+    UIManager:broadcastEvent(Event:new("UpdateFooter", true))
+end
+
 function ReadTimer:unschedule()
     if self:scheduled() then
         UIManager:unschedule(self.alarm_callback)
         self.time = 0
     end
+    UIManager:unschedule(self.update_status_bars, self)
 end
 
 function ReadTimer:rescheduleIn(seconds)
     self.time = os.time() + seconds
     UIManager:scheduleIn(seconds, self.alarm_callback)
+    if self.show_value_in_header or self.show_value_in_footer then
+        self:update_status_bars(seconds)
+    end
+end
+
+function ReadTimer:addCheckboxes(widget)
+    local checkbox_title = TextBoxWidget:new{
+        alignment = "left",
+        text = _("Display timer value in:"),
+        face = Font:getFace("smallinfofont"),
+        width = widget:getAddedWidgetAvailableWidth(),
+    }
+    local checkbox_header = CheckButton:new{
+        text = _("top status bar"),
+        checked = self.show_value_in_header,
+        parent = widget,
+        callback = function()
+            self.show_value_in_header = not self.show_value_in_header
+            G_reader_settings:saveSetting("readtimer_show_value_in_header", self.show_value_in_header)
+            if self.show_value_in_header then
+                self:addAdditionalHeaderContent()
+            else
+                self:removeAdditionalHeaderContent()
+            end
+        end,
+    }
+    local checkbox_footer = CheckButton:new{
+        text = _("bottom status bar"),
+        checked = self.show_value_in_footer,
+        parent = widget,
+        callback = function()
+            self.show_value_in_footer = not self.show_value_in_footer
+            G_reader_settings:saveSetting("readtimer_show_value_in_footer", self.show_value_in_footer)
+            if self.show_value_in_footer then
+                self:addAdditionalFooterContent()
+            else
+                self:removeAdditionalFooterContent()
+            end
+        end,
+    }
+    local checkbox_frame = HorizontalGroup:new{
+        align = "center",
+        HorizontalSpan:new{ width = Size.span.horizontal_default },
+        VerticalGroup:new{
+            checkbox_title,
+            checkbox_header,
+            checkbox_footer,
+        },
+    }
+    widget:addWidget(checkbox_frame)
```

Comment:
I propose 2 check buttons added separately as we do everywhere (the API needs fixing).
"Show timer in status bar"
"Show timer in alt status bar"